### PR TITLE
Added Categorical distribution

### PR DESCRIPTION
**Pull request by fonnesbeck**
Implemented categorical distribution and associated test.

### CODE REVIEW

**Comment by fonnesbeck** [comment]
No clue why it fails in TravisCI. It passes locally; can anyone else reproduce?

**Comment by fonnesbeck** [comment]
OK, runs clean on Travis now. Please have a look.

File: pymc/distributions/discrete.py [hunk 1]
```diff
@@ -1,7 +1,9 @@
 from .dist_math import *
+import theano
```
Comment (jsalvatier) [inline pymc/distributions/discrete.py:2]: Doesn't seem to get used.

**Comment by jsalvatier** [comment]
Looks good.

**Comment by fonnesbeck** [comment]
Thanks. Will remove the import in master.

**Pull request merged**